aws/resource_aws_instance.go

```diff
@@ -655,21 +655,21 @@ func resourceAwsInstanceCreate(d *schema.ResourceData, meta interface{}) error {
 	}
 
 	instance := runResp.Instances[0]
-	log.Printf("[INFO] Instance ID: %s", *instance.InstanceId)
+	log.Printf("[INFO] Instance ID: %s", aws.StringValue(instance.InstanceId))
 
 	// Store the resulting ID so we can look this up later
-	d.SetId(*instance.InstanceId)
+	d.SetId(aws.StringValue(instance.InstanceId))
 
 	// Wait for the instance to become running so we can get some attributes
 	// that aren't available until later.
 	log.Printf(
 		"[DEBUG] Waiting for instance (%s) to become running",
-		*instance.InstanceId)
+		aws.StringValue(instance.InstanceId))
 
 	stateConf := &resource.StateChangeConf{
 		Pending:    []string{"pending"},
```

Comment:
For the instance state name values you could use the Go SDK constants

```go
const (
	// InstanceStateNamePending is a InstanceStateName enum value
	InstanceStateNamePending = "pending"

	// InstanceStateNameRunning is a InstanceStateName enum value
	InstanceStateNameRunning = "running"

	// InstanceStateNameShuttingDown is a InstanceStateName enum value
	InstanceStateNameShuttingDown = "shutting-down"

	// InstanceStateNameTerminated is a InstanceStateName enum value
	InstanceStateNameTerminated = "terminated"

	// InstanceStateNameStopping is a InstanceStateName enum value
	InstanceStateNameStopping = "stopping"

	// InstanceStateNameStopped is a InstanceStateName enum value
	InstanceStateNameStopped = "stopped"
)
```

defined in `github.com/aws/aws-sdk-go/service/ec2/api.go`.

Reply:
nice, thanks @ewbankkit! I'll update those